pytorch_tabnet/metrics.py

```diff
@@ -24,20 +24,26 @@ def UnsupervisedLoss(y_pred, embedded_x, obf_vars, eps=1e-9):
         Orginal input embedded by network
     obf_vars : torch.Tensor
         Binary mask for obfuscated variables.
-        1 means the variables was obfuscated so reconstruction is based on this.
+        1 means the variable was obfuscated so reconstruction is based on this.
     eps : float
         A small floating point to avoid ZeroDivisionError
         This can happen in degenerated case when a feature has only one value
 
     Returns
     -------
     loss : torch float
-        Unsupervised loss, average value over batch samples.    
+        Unsupervised loss, average value over batch samples.
     """
     errors = y_pred - embedded_x
-    reconstruction_errors = torch.mul(errors, obf_vars)**2
-    batch_stds = torch.std(embedded_x, dim=0)**2 + eps
+    reconstruction_errors = torch.mul(errors, obf_vars) ** 2
+    batch_stds = torch.std(embedded_x, dim=0) ** 2 + eps
     features_loss = torch.matmul(reconstruction_errors, 1 / batch_stds)
+    # compute the number of non-obfuscated variables
+    nb_used_variables = torch.sum(1 - obf_vars, dim=1)
+    # print(nb_used_variables)
+    # take the mean of the used variable errors
+    # print(features_loss)
+    features_loss = features_loss / (1 + nb_used_variables)
```

Comment:
```python
    features_loss = features_loss / (1 + nb_masked_variables)
```

Reply:
why is there a 1 here? shouldn't this be an epsilon? So that we avoid 0 division error but don't change the results when there are masked_variables?

Reply:
I still think that we should divide by `epsilon + nb_reconstructed_features` it's safe to do so since when `nb_reconstructed_features=0` then the numerator should be zero as well. So the loss won't blow up.